torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1069,6 +1276,7 @@ def _rebuild_full_params(self) -> None:
         """
         Gather all shards of params.
         """
+        self._lazy_init()
```

Comment:
Seems unrelated, why do we need to add `_lazy_init` in _rebuild_full_params now?

Reply:
_rebuild_full_params are used in the several our tests. In the tests, there are no guarantee that forward will be called. Adding the lazy_init should make the code more robust.